pkg/store/mockstore/unistore/tikv/util_test.go

```diff
@@ -0,0 +1,432 @@
+			current:  []byte("abc"),
+			endKey:   []byte("abc"),
+			expected: true,
+		},
+		{
+			name:     "current greater than end key",
+			current:  []byte("bcd"),
+			endKey:   []byte("abc"),
+			expected: true,
+		},
+		{
+			name:     "current less than end key",
+			current:  []byte("abc"),
+			endKey:   []byte("bcd"),
+			expected: false,
+		},
+		{
+			name:     "current empty, end key not empty",
+			current:  []byte{},
+			endKey:   []byte("abc"),
+			expected: false,
+		},
+		{
+			name:     "both empty",
+			current:  []byte{},
+			endKey:   []byte{},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := exceedEndKey(tt.current, tt.endKey)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestSortAndDedupHashVals(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []uint64
+		expected []uint64
+	}{
+		{
+			name:     "empty slice",
+			input:    []uint64{},
+			expected: []uint64{},
+		},
+		{
+			name:     "single element",
+			input:    []uint64{1},
+			expected: []uint64{1},
+		},
+		{
+			name:     "already sorted without duplicates",
+			input:    []uint64{1, 2, 3, 4, 5},
+			expected: []uint64{1, 2, 3, 4, 5},
+		},
+		{
+			name:     "unsorted without duplicates",
+			input:    []uint64{5, 3, 1, 4, 2},
+			expected: []uint64{1, 2, 3, 4, 5},
+		},
+		{
+			name:     "with duplicates",
+			input:    []uint64{3, 1, 4, 1, 5, 9, 2, 6, 5, 3},
+			expected: []uint64{1, 2, 3, 4, 5, 6, 9},
+		},
+		{
+			name:     "all same elements",
+			input:    []uint64{7, 7, 7, 7},
+			expected: []uint64{7},
+		},
+		{
+			name:     "two elements same",
+			input:    []uint64{3, 3},
+			expected: []uint64{3},
+		},
+		{
+			name:     "two different elements",
+			input:    []uint64{3, 1},
+			expected: []uint64{1, 3},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Make a copy to avoid modifying the original test data
+			input := make([]uint64, len(tt.input))
+			copy(input, tt.input)
+
+			result := sortAndDedupHashVals(input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestMutationsToHashVals(t *testing.T) {
+	tests := []struct {
+		name      string
+		mutations []*kvrpcpb.Mutation
+		expected  []uint64
+	}{
+		{
+			name:      "empty mutations",
+			mutations: []*kvrpcpb.Mutation{},
+			expected:  []uint64{},
+		},
+		{
+			name: "single mutation",
+			mutations: []*kvrpcpb.Mutation{
+				{Key: []byte("key1")},
+			},
+			expected: []uint64{farm.Fingerprint64([]byte("key1"))},
+		},
+		{
+			name: "multiple mutations without duplicates",
+			mutations: []*kvrpcpb.Mutation{
+				{Key: []byte("key1")},
+				{Key: []byte("key2")},
+				{Key: []byte("key3")},
+			},
+			expected: func() []uint64 {
+				vals := []uint64{
+					farm.Fingerprint64([]byte("key1")),
+					farm.Fingerprint64([]byte("key2")),
+					farm.Fingerprint64([]byte("key3")),
+				}
+				return sortAndDedupHashVals(vals)
```

Comment:
because these tests calls `sortAndDedupHashVals`, it doesn't check the logic inside `sortAndDedupHashVals`. I think you can delete them. Only `TestSortAndDedupHashVals` is enough